packages/orbit-components/src/Stack/__tests__/index.test.js

```diff
@@ -1,480 +1,163 @@
-      <Button>Sign In</Button>
-    </Stack>,
-  );
-  const stack = component.find("Stack__StyledStack");
-  it("should contain default styles", () => {
-    expect(stack).toHaveStyleRule("flex-direction", "row");
-    expect(stack).toHaveStyleRule("margin", "0 4px 0 0!important", {
-      modifier: "& > *",
-    });
-  });
-  it("should contain mediumMobile styles", () => {
-    expect(stack).toHaveStyleRule("flex-direction", "column-reverse", {
-      media: getBreakpointWidth(QUERIES.MEDIUMMOBILE, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("margin", "4px 0 0 0!important", {
-      media: getBreakpointWidth(QUERIES.MEDIUMMOBILE, defaultTheme),
-      modifier: "& > *",
-    });
-  });
-  it("should contain largeMobile styles", () => {
-    expect(stack).toHaveStyleRule("margin", "32px 0 0 0!important", {
-      media: getBreakpointWidth(QUERIES.LARGEMOBILE, defaultTheme),
-      modifier: "& > *",
-    });
-  });
-  it("should contain tablet styles", () => {
-    expect(stack).toHaveStyleRule("flex-direction", "row-reverse", {
-      media: getBreakpointWidth(QUERIES.TABLET, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("margin", "0 0 0 32px!important", {
-      media: getBreakpointWidth(QUERIES.TABLET, defaultTheme),
-      modifier: "& > *",
-    });
-  });
-  it("should contain desktop styles", () => {
-    expect(stack).toHaveStyleRule("margin", "0 0 0 2px!important", {
-      media: getBreakpointWidth(QUERIES.DESKTOP, defaultTheme),
-      modifier: "& > *",
-    });
-  });
-  it("should contain largeDesktop styles", () => {
-    expect(stack).toHaveStyleRule("flex-direction", "column", {
-      media: getBreakpointWidth(QUERIES.LARGEDESKTOP, defaultTheme),
-    });
-    expect(stack).toHaveStyleRule("margin", "0 0 2px 0!important", {
-      media: getBreakpointWidth(QUERIES.LARGEDESKTOP, defaultTheme),
-      modifier: "& > *",
-    });
+  it.each(Object.entries(QUERIES))("should have styles for each mediaquery: %s", (name, query) => {
+    const dataTest = `${query}-test`;
+
+    render(
+      <Stack
+        dataTest={dataTest}
+        direction={DIRECTIONS.ROW}
+        spacing={SPACINGS.NONE}
+        mediumMobile={{ spacing: SPACINGS.XXSMALL }}
+        largeMobile={{ spacing: SPACINGS.XSMALL }}
+        tablet={{ spacing: SPACINGS.SMALL }}
+        desktop={{ spacing: SPACINGS.MEDIUM }}
+        largeDesktop={{ spacing: SPACINGS.LARGE }}
+      >
+        <InputField type="password" label="Password" help="You need some help!" />
+        <Button>Sign In</Button>
+      </Stack>,
+    );
+
+    const expected = {
+      mediumMobile: `${theme.orbit.spaceXXSmall}!important`,
+      largeMobile: `${theme.orbit.spaceXSmall}!important`,
+      tablet: `${theme.orbit.spaceSmall}!important`,
+      desktop: `${theme.orbit.spaceMedium}!important`,
+      largeDesktop: `${theme.orbit.spaceLarge}!important`,
+    };
+
+    expect(screen.getByTestId(dataTest)).toHaveStyle({ marginRight: expected[query] });
```

Comment:
The use of `it.each` here suggests different JSX for each iteration, which is not true (except for `dataTest`), so it looks like you can just do a `forEach` loop for these assertions instead?